### PR TITLE
Refactor the frontend testing to use the fake database backend

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Fixes https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/6118707

### What this PR does / why we need it:

Refactors the frontend unit tests to use the fake database backend.

### Test plan for issue:

Mostly test updates.

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

**Comment by hawkowl** [comment]
@jim-minter this should be good for a look now

File: test/database/check.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	"errors"
+	"fmt"
+
+	"github.com/go-test/deep"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+)
+
+const DELETION_TIME_SET = 123456789
```
Comment (mjudeikis) [inline test/database/check.go:17]: Any reason this is ALL Caps?

**Comment by mjudeikis** [comment]
@jim-minter This looks ok, but I lost track of all the parts involved in this. Would be great if you can take a look.

File: test/database/check.go (hunk 1 above)
Comment (hawkowl) [reply test/database/check.go:17]: To distinguish it as a const, I guess!

Comment (mjudeikis) [reply test/database/check.go:17]: sound like Python :) lets keep it go :D

**Review by mjudeikis — APPROVED** [review]
:eyes:

**Pull request merged**